model/countries/FR/FR-parsing-rules.yaml

```diff
@@ -0,0 +1,120 @@
+      - capture:
+          output: street
+          parts: [ {regex_reference: kMultipleLazyWordsRe} ]
+
+
+  StreetAddressDecomposition:
+    capture:
+      output: street-address-alternative-1
+      parts:
+      - capture:
+          output: address-overflow
+          parts: [ {regex_fragment: '(?:[^,\r\n]+)'} ]
+          quantifier: MATCH_OPTIONAL
+      - separator: {regex_reference: kCommaOrWhitespaceSeparator}
+      - capture_reference: ParseBuildingLocation
+
+
+
+parsing_definitions:
+  building-location:
+    decomposition:
+      capture_reference: ParseBuildingLocation
+  street-address-alternative-1:
+    decomposition:
+      capture_reference: StreetAddressDecomposition
+
+test_parsing_definitions:
+- id: "Test 1"
+  type: building-location
+  input: "1661 Place Charles de Gaulle"
+  output:
+    street: "Place Charles de Gaulle"
+    building: "1661"
+- id: "Test 2"
+  type: building-location
+  input: "58 rue du Gue Jacquet"
+  output:
+    street: "rue du Gue Jacquet"
+    building: "58"
+- id: "Test 3"
+  type: street-address-alternative-1
+  input: "58 rue du Gue Jacquet"
+  output:
+    street-address-alternative-1: "58 rue du Gue Jacquet"
+    building-location: "58 rue du Gue Jacquet"
+    street: "rue du Gue Jacquet"
+    building: "58"
+- id: "Test 4"
+  type: street-address-alternative-1
+  input: "1661 Place Charles de Gaulle"
+  output:
+    street-address-alternative-1: "1661 Place Charles de Gaulle"
+    building-location: "1661 Place Charles de Gaulle"
+    street: "Place Charles de Gaulle"
+    building: "1661"
+- id: "Test 5"
+  type: street-address-alternative-1
+  input: "Appartament 36\n1661 Place Charles de Gaulle"
+  output:
+    street-address-alternative-1: "Appartament 36\n1661 Place Charles de Gaulle"
+    building-location: "1661 Place Charles de Gaulle"
+    street: "Place Charles de Gaulle"
+    building: "1661"
+    address-overflow: "Appartament 36"
+- id: "Test 6"
+  type: street-address-alternative-1
+  input: "Appartament 1\n58 rue du Gue Jacquet"
+  output:
+    street-address-alternative-1: "Appartament 1\n58 rue du Gue Jacquet"
+    building-location: "58 rue du Gue Jacquet"
+    street: "rue du Gue Jacquet"
+    building: "58"
+    address-overflow: "Appartament 1"
+- id: "Test 7"
+  type: street-address-alternative-1
+  input: "Appartament 36, 1661 Place Charles de Gaulle"
+  output:
+    street-address-alternative-1: "Appartament 36, 1661 Place Charles de Gaulle"
+    building-location: "1661 Place Charles de Gaulle"
+    street: "Place Charles de Gaulle"
+    building: "1661"
+    address-overflow: "Appartament 36"
+- id: "Test 8"
+  type: street-address-alternative-1
+  input: "Appartament 1, 58 rue du Gue Jacquet"
+  output:
+    street-address-alternative-1: "Appartament 1, 58 rue du Gue Jacquet"
+    building-location: "58 rue du Gue Jacquet"
+    street: "rue du Gue Jacquet"
+    building: "58"
+    address-overflow: "Appartament 1"
```

Comment:
Can we add tests for when the street address contains things like Battiment or building name
- Apt. 45, Résidence Les Lilas, 12 Rue de la Paix
- Bâtiment A, 5 Avenue des Champs-Élysées, 75008 Paris, France